#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_latest_highlighting_pattern.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: This is an update to the latest filetypes list and highlighting pattern
## DP: of the bluefish stable tree.
## DP:
## DP: data/filetypes.in  1.19.2.6
## DP: data/highlighting.in  1.52.2.42
## DP:
## DP: https://bugs.launchpad.net/ubuntu/+source/bluefish/+bug/228068

@DPATCH@
diff -urNad trunk~/data/filetypes.in trunk/data/filetypes.in
--- trunk~/data/filetypes.in	2006-06-01 12:12:07.000000000 +0200
+++ trunk/data/filetypes.in	2008-05-09 00:15:14.000000000 +0200
@@ -1,25 +1,27 @@
 text:.txt:::1::0:
-php:.php\:.php4\:.inc\:.php3: <>'"/?$\\t-{}[]{}\n;:PKGDATADIRicon_php.png:1:^(#!/[a-z/]+php|<\\?php):2:
 html:.html\:.htm\:.shtml\:.shtm:<> \n":PKGDATADIRicon_html.png:1:<html[^>]*>:2:
-python:.py::PKGDATADIRicon_python.png:1:^#!/[a-z/]+/python:0:
-c:.c\:.h:(){}'[]\n/#" ;:PKGDATADIRicon_c.png:1::0:
-java:.java:(){}'[]\n/" :PKGDATADIRicon_java.png:1::0:
-perl:.pl\:.PL\:.cgi\:.pm:::1:^#!/[a-z/]+perl:0:
-objectfile:.o\:.class:::0::0:
+php:.php\:.php4\:.inc\:.php3: <>'"/?$\\t-{}[]{}\n;:PKGDATADIRicon_php.png:1:^(#!/[a-z/]+php|<\\?php):2:
 stylesheet:.css:::1::0:
-webimage:.jpg\:.png\:.gif\:.jpeg::PKGDATADIRicon_image.png:0::0:
-image:.jpg\:.png\:.gif\:.jpeg\:.tif\:.tiff\:.xpm\:.xcf::PKGDATADIRicon_image.png:0::0:
-sql:.sql:(){}'[]\n" ::1::0:
-jsp:.jsp:(){}'[]\n/#" :PKGDATADIRicon_java.png:1:^#!/[a-z/]+jsp:0:
-xml:.xml\:.sgml\:.xsl:<> \n":PKGDATADIRicon_xml.png:1:<\\?xml:1:
-javascript:.js:(){}'[]\n/" ::1::0:
+c:.c\:.cpp\:.cc\:.h:(){}'[]\n/#" ;:PKGDATADIRicon_c.png:1::0:
 cfml:.cfm: <>'"/?$\\t-{}[]{}\n;:PKGDATADIRicon_cfml.png:1::2:
-pascal:.pas: (){}'[]\n;":PKGDATADIRicon_pascal.png:1::0:
-R:.r\:.rd\:.R\:.Rd\:.RD: <>=+-*/'"{}[]()\t\n.,\:$&|#:PKGDATADIRicon_r.png:1::0:
+changelog:ChangeLog::::::
+D:.d\:.di:(){}'[]\n/#" ;::1::0:
 gettext po:.po:"\n::1::0:
+java:.java:(){}'[]\n/" :PKGDATADIRicon_java.png:1::0:
+javascript:.js:(){}'[]\n/" ::1::0:
+jsp:.jsp:(){}'[]\n/#" :PKGDATADIRicon_java.png:1:^#!/[a-z/]+jsp:0:
+kid:.kid:<> \\n":PKGDATADIRicon_xml.png::::
 octave:.m:::1:^#!/[a-z/]+/octave:0:
+pascal:.pas: (){}'[]\n;":PKGDATADIRicon_pascal.png:1::0:
+perl:.pl\:.PL\:.cgi\:.pm:::1:^#!/[a-z/]+perl:0:
+python:.py::PKGDATADIRicon_python.png:1:^#!/[a-z/]+/python:0:
+R:.r\:.rd\:.R\:.Rd\:.RD: <>=+-*/'"{}[]()\t\n.,\:$&|#:PKGDATADIRicon_r.png:1::0:
+Ruby:.rb:::1::0:
 shell:.sh:::1:^#!/[a-z/]+/(ba)?sh$:0:
+sql:.sql:(){}'[]\n" ::1::0:
 tcl:.tcl:::1::0:
-Ruby:.rb:::1::0:
+xml:.xml\:.sgml\:.xsl:<> \n":PKGDATADIRicon_xml.png:1:<\\?xml:1:
+webimage:.jpg\:.png\:.gif\:.jpeg::PKGDATADIRicon_image.png:0::0:
+image:.jpg\:.png\:.gif\:.jpeg\:.tif\:.tiff\:.xpm\:.xcf::PKGDATADIRicon_image.png:0::0:
+objectfile:.o\:.class:::0::0:
 bfproject:.bfproject:::0::0:
-changelog:ChangeLog::::::
diff -urNad trunk~/data/highlighting.in trunk/data/highlighting.in
--- trunk~/data/highlighting.in	2006-09-26 19:29:32.000000000 +0200
+++ trunk/data/highlighting.in	2008-05-09 00:15:32.000000000 +0200
@@ -5,10 +5,10 @@
 c:Comment (C99/C++):0://.*?$::2:^(top|preprocessor)$:#0000FF::1:2:
 c:Keywords:0:\\b(break|case|continue|default|do|else|for|goto|if|return|sizeof|switch|while)\\b::2::#009999::2:0:
 c:Storage Keywords:0:\\b(auto|char|const|double|enum|extern|FILE|float|int(8|16|32|64)?|long|register|short|signed|size_t|static|struct|typedef|union|unsigned|void|volatile)\\b::2::#000099::2:0:
-c:GTK Keywords:0:\\b(Gtk(Widget|AccelLabel|AccelGroup|AccelMap|Accessible|Action|ActionEntry|ActionGroup|Adjustment|Alignment|Allocation|Arrow|ArrowType|AspectFrame|Bin|Border|Box|Button|ButtonBox|ButtonsType|Calendar|CellEditable|CellEditableIface|CellRenderer|CellRenderPixbuf|CellRendererState|CellRendererText|CellRendererToggle|CheckButton|Clipboard|CList|ColorButton|ColorSelection|ColorSelectionDialog|Combo|ComboBox|ComboBoxEntry|Container|CTree|Curve|Dialog|DrawingArea|Editable|Entry|EntryCompletion|EventBox|Expander|ExpanderStyle|FileChooserAction|FileFilter|FileFilterFlags|FileFilterFunc|FileSelectionDialog|Fixed|FontButton|FontSelection|FontSelectionDialog|Frame|GammaCurve|HandleBox|HBox|HButtonBox|HPaned|HRuler|HScale|HScrollbar|HSeparator|IconFactory|IconSet|IconSource|Image|ImageMenuItem|IMContext|IMContextSimple|IMMulticontext|InputDialog|Invisible|ItemFactory|ItemFactoryEntry|Label|Layout|List|ListItem|ListStore|Menu|MenuBar|MenuItem|MenuShell|MessageDialog|MessageType|Misc|Notebook|Object|OldEditable|OptionMenu|Orientation|Paned|Pixmap|Plug|PositionType|Preview|Progress|ProgressBar|RadioActionEntry|RadioButton|RadioMenuItem|Range|Requisition|ResponseType|RcProperty|RcPropertyParser|RcStyle|Ruler|Scale|Scrollbar|ScrolledWindow|Separator|SeparatorMenuItem|Settings|ShadowType|SignalFunc|SizeGroup|Socket|SpinButton|StateType|StatusBar|Style|Table|TearoffMenuItem|Text|TextBuffer|TextChildAnchor|TextIter|TextMark|TextTag|TextTagTable|TextView|TextWindowType|TipsQuery|ToggleActionEntry|ToggleButton|Toolbar|ToolItem|Tooltips|Tree|TreeDragDest|TreeDragSource|TreeItem|TreeIter|TreeModel|TreeModelIface|TreeModelSort|TreePath|TreeRowReference|TreeSelection|TreeSortable|TreeStore|TreeView|TreeViewColumn|UIManager|VBox|VButtonBox|Viewport|VPaned|VRuler|VScale|VScrollbar|VSeparator|Window|WindowGroup|WindowPosition|WrapMode))\\b::2::#990000::2:0:
+c:GTK Keywords:0:\\b(Gtk(Widget|AccelLabel|AccelGroup|AccelMap|Accessible|Action|ActionEntry|ActionGroup|Adjustment|Alignment|Allocation|Arrow|ArrowType|AspectFrame|Bin|Border|Box|Button|ButtonBox|ButtonsType|Calendar|CellEditable|CellEditableIface|CellRenderer|CellRenderPixbuf|CellRendererState|CellRendererText|CellRendererToggle|CheckButton|Clipboard|CList|ColorButton|ColorSelection|ColorSelectionDialog|Combo|ComboBox|ComboBoxEntry|Container|CTree|Curve|Dialog|DrawingArea|Editable|Entry|EntryCompletion|EventBox|Expander|ExpanderStyle|FileChooserAction|FileFilter|FileFilterFlags|FileFilterFunc|FileSelectionDialog|Fixed|FontButton|FontSelection|FontSelectionDialog|Frame|GammaCurve|HandleBox|HBox|HButtonBox|HPaned|HRuler|HScale|HScrollbar|HSeparator|IconFactory|IconSet|IconSize|IconSource|Image|ImageMenuItem|IMContext|IMContextSimple|IMMulticontext|InputDialog|Invisible|ItemFactory|ItemFactoryEntry|Label|Layout|List|ListItem|ListStore|Menu|MenuBar|MenuItem|MenuShell|MessageDialog|MessageType|Misc|Notebook|Object|OldEditable|OptionMenu|Orientation|Paned|Pixmap|Plug|PositionType|Preview|Progress|ProgressBar|RadioActionEntry|RadioButton|RadioMenuItem|Range|Requisition|ResponseType|RcProperty|RcPropertyParser|RcStyle|Ruler|Scale|Scrollbar|ScrolledWindow|Separator|SeparatorMenuItem|Settings|ShadowType|SignalFunc|SizeGroup|Socket|SpinButton|StateType|StatusBar|Style|Table|TearoffMenuItem|Text|TextBuffer|TextChildAnchor|TextIter|TextMark|TextTag|TextTagTable|TextView|TextWindowType|TipsQuery|ToggleActionEntry|ToggleButton|Toolbar|ToolItem|Tooltips|Tree|TreeDragDest|TreeDragSource|TreeItem|TreeIter|TreeModel|TreeModelIface|TreeModelSort|TreePath|TreeRowReference|TreeSelection|TreeSortable|TreeStore|TreeView|TreeViewColumn|UIManager|VBox|VButtonBox|Viewport|VPaned|VRuler|VScale|VScrollbar|VSeparator|Window|WindowGroup|WindowPosition|WrapMode))\\b::2::#990000::2:0:
 c:GTK Widget Class:0:\\b(Gtk([a-zA-Z]+)Class)\\b::2::#990000::2:0:
 c:Glib Basictypes:0:\\b(gu?int(8|16|32|64)?|gshort|gushort|glong|gulong|gfloat|gdouble|gchar|guchar|gunichar|gpointer|gconstpointer|gboolean|gsize|gssize)\\b::2::#990000::2:0:
-c:Glib Keywords:0:\\b(GArray|GCallback|GClassInitFunc|GClosure|GCompareDataFunc|GCompareFunc|GDate|GDir|GEnumValue|GError|GErrorType|GFlagsValue|GFunc|GHashTable|GInstanceInitFunc|GIOChannel|GIOStatus|GList|GMarkupParseFlags|GMarkupParser|GMarkupParseContext|GObject|GObjectClass|GObjectConstructParam|GParamSpec|GPatternSpec|GPtrArray|GQuark|GSignalAccumulator|GScanner|GSList|GString|GTimer|GTokenType|GTokenValue|GTree|GType|GTypeInfo|GValue)\\b::2::#990000::2:0:
+c:Glib Keywords:0:\\b(GArray|GCallback|GClassInitFunc|GClosure|GCompareDataFunc|GCompareFunc|GDate|GDir|GEnumValue|GError|GErrorType|GFlagsValue|GFunc|GHashTable|GInstanceInitFunc|GIOChannel|GIOStatus|GKeyFile|GList|GMarkupParseFlags|GMarkupParser|GMarkupParseContext|GObject|GObjectClass|GObjectConstructParam|GParamSpec|GPatternSpec|GPtrArray|GQuark|GSignalAccumulator|GScanner|GSList|GString|GTimer|GTokenType|GTokenValue|GTree|GType|GTypeInfo|GValue)\\b::2::#990000::2:0:
 c:GDK Keywords:0:\\b(Gdk(Color|Colormap|CursorType|Display|DragContext|Drawable|Event|EventAny|EventButton|EventCrossing|EventExpose|EventFocus|EventKey|EventScroll|Font|Pixbuf|Pixdata|Pixmap|Point|Rectangle|Region|Screen|Window|WindowAttr))\\b::2::#990000::2:0:
 c:GnomeVFS Keywords:0:\\b(GnomeVFS(AsyncHandle|DirectoryHandle|FileInfo|FileSize|FileType|Handle|Result|ToplevelURI|URI|XferProgressInfo))\\b::2::#990000::2:0:
 c:Braces:0:[{}]::2::#000000::2:0:
@@ -35,6 +35,23 @@
 cfml:HTML Comment:0:<!--:-->:1::#AAAAAA::1:2:
 cfml:Special Characters:1:&[^;]*;::2::#999999::2:0:
 cfml:DocType:1:<![a-z0-9]+:[^?-]>:1::#bb8800::0:0:
+D:Decimal:0:\\b-?[0-9_]+(L[uU]?|[uU]L?)?\\b::2::#009900::0:0:
+D:Hex:0:\\b0[xX][A-Fa-f0-9_]+(L[uU]?|[uU]L?)?\\b::2::#009900::0:0:
+D:Binary:0:\\b0[bB][01_]+(L[uU]?|[uU]L?)?\\b::2::#009900::0:0:
+D:Float:0:\\b[0-9_]+(\\.[0-9_]+)?([eE][-+]?[0-9_]+)?[fFL]?i?\\b::2::#009900::0:0:
+D:HexFloat:0:\\b0[xX][A-Fa-f0-9_]+(\\.[A-Fa-f0-9_]+)?([pP][-+]?[0-9_]+)?[fFL]?i?\\b::2::#009900::0:0:
+D:Keywords:1:\\b(abstract|alias|align|asm|assert|auto|body|break|case|cast|catch|class|const|continue|debug|default|delete|deprecated|do|else|enum|export|extern|final|finally|for|foreach|foreach_reverse|goto|if|import|in|inout|interface|invariant|is|lazy|mixin|module|new|out|override|package|pragma|private|protected|public|return|scope|static|struct|switch|synchronized|template|throw|try|typedef|typeid|typeof|union|unittest|ushort|version|volatile|while|with)\\b::2::#009999::2:0:
+D:Basic Types:1:\\b(delegate|function|bit|bool|byte|ubyte|short|ushort|int|uint|long|ulong|cent|char|wchar|dchar|float|double|real|ifloat|idouble|ireal|cfloat|cdouble|creal|void)\\b::2::#000099::2:0:
+D:Special Values:1:\\b(true|false|null|this|super)\\b::2::#999900::2:0:
+D:Braces:0:[{}]::2::#000000::2:0:
+D:Character:0:'.*?'::2::#009900::0:0:
+D:Character Escape:0:\\\\.::2:^character$:#009900::0:0:
+D:String (wysiwyg):0:(r".*?"[cwd]?)|(`.*?`[cwd]?)::2::#009900::0:0:
+D:String Escape:0:\\\\(n|t|v|b|r|f|a|\\\\|\\?|'|"|\\d{1,3}|x\\d+)::2:^String$:#990099::0:0:
+D:String:0:x?"((\\\\")|[^"])*"[cwd]?::2::#009900::0:0:
+D:Comment (single line):0://.*?$::2::#0000FF::1:2:
+D:Comment (multiline):0:/\\*.*?\\*/::2::#0000FF::1:2:
+D:Comment (nested):0:/\\+((?R)|.)*?\\+/::2::#0000FF::1:2:
 gettext po:Untranslated:1:msgstr(?\:\\[[0-9]+\\])? ""\\n\\n::2::#990000:#ffffff:0:0:
 gettext po:Text:1:":":1::#009900:#ffffff:0:2:
 gettext po:Todo:1:^#,.*?$::2::#ffffff:#990000:2:0:
@@ -105,6 +122,43 @@
 jsp:HTML Table Tags:1:^(/)?(table|td|tr|tbody)$::2:^<html> Tags$:#5005AA::0:0:
 jsp:<html> Tags:1:1::3:^HTML$:#550044::2:0:
 jsp:HTML:1:<((/)?[a-z0-9]+):>:1::#000077::0:0:
+kid:Keywords Definition:0:(class|def)[\\t ]+[a-zA-Z0-9_]+::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#0000AA::2:0:
+kid:Operators:0:[\\-\\+/\\*\:.,;(){}\\[\\]<>=]::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#000088::0:0:
+kid:Keywords:0:\\b(and|break|continue|del|elif|else|exec|for|if|in|is|not|or|pass|print|return|while)\\b::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#000000::2:0:
+kid:Keywords Self:0:(self\\.?)::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#000088::2:0:
+kid:Numbers:1:[+\\-]?[0-9][0-9xa-f]*(\\.[0-9a-f]+(e[0-9]+)?)?::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:red::0:0:
+kid:Keywords Exception:0:\\b((assert|except|finally|try)[^\:]*\:|raise[^\\n]*\\n)::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#DD0000::2:0:
+kid:Include:0:^(import |from ).*?$::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#990099::0:0:
+kid:Identifiers:1:[a-zA-Z_][a-zA-Z0-9_]*::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:brown::0:0:
+kid:Comment:0:#.*?$::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#AAAAAA::1:2:
+kid:Multi-line Comment:0:''':''':1:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#AAAAAA::1:2:
+kid:Multi-line Comment2:0:""":""":1:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#AAAAAA::1:2:
+kid:String Double:0:":":1:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#009900::0:0:
+kid:String Double Escape:0:\\\\.::2:^string-double$:#009900::0:0:
+kid:String Single:0:':':1:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#009900::0:0:
+kid:String Single Escape:0:\\\\.::2:^string-single$:#009900::0:0:
+kid:Special Values:0:\\bNone\\b::2:^(PyAttribute Content|FormatPython|PyEmbeddedContent|PyEmbeddedExpression)$:#999900::0:0:
+kid:Braces:0:[{()}\\[\\]]::2::#000000::2:
+kid:FormatPython:1:<\\?python:\\?>:1:^top$:::0:0:
+kid:Format:1:<\\?:\\?>:1:^top$:#000000::2:1:
+kid:Tag:1:<((/)?[a-z0-9\:-]+):[^?-]?>:1::::0:0:
+kid:Tag Name:1:1::3:^Tag$:#005676::2:0:
+kid:PyAttributeApos:0:py\\\:(for|if|content|replace|strip|attrs|def|match|extends|layout)='([^']*)'::2:^Tag$:::1:0:
+kid:PyAttributeQuot:0:py\\\:(for|if|content|replace|strip|attrs|def|match|extends|layout)="([^"]*)"::2:^Tag$:::1:0:
+kid:PyAttribute Name:1:1::3:^(PyAttributeApos|PyAttributeQuot)$:red:yellow:1:0:
+kid:PyAttribute Content:1:2::3:^(PyAttributeApos|PyAttributeQuot)$:::1:0:
+kid:AttributeApos:1:([a-z\:\\-]*=)'([^']*)'::2:^Tag$:::0:0:
+kid:AttributeQuot:1:([a-z\:\\-]*=)"([^"]*)"::2:^Tag$:::0:0:
+kid:Attribute Name:1:1::3:^(AttributeApos|AttributeQuot)$:#ff7600::1:0:
+kid:Attribute Content:1:2::3:^(AttributeApos|AttributeQuot)$:::0:0:
+kid:DocType:1:<!DOCTYPE [a-z0-9]+:>:1::#b35300::0:0:
+kid:Entity:1:[\\[]:[\\]]:1:^DocType$:#b35300::2:1:
+kid:Comment:0:<!--:-->:1:^top|DocType|Entity$:#AAAAAA::1:2:
+kid:Character Data:1:<!\\[CDATA\\[:\\]\\]>:1::#a2b300::0:0:
+kid:Character Entity:1:&[^;]+;::2::#9b0054::0:0:
+kid:PyEmbedded:0:\\$([a-zA-Z_][a-zA-Z_0-9\\.]*)::2:^(Attribute Content|top)$:::0:0:
+kid:PyEmbeddedContent:0:1::3:^(PyEmbedded|PyEmbeddedExpression1)$:::0:0:
+kid:PyEmbeddedExpression1:0:\\${([^}]*)}::2:^(Attribute Content|top)$:::0:0:
 octave:String (escaped):1:\\\\[\\\\"'0abfnrtv]::2:^String$:#000099::0:0:
 octave:String:1:":":1:^(top|Function)$:#009900::0:0:
 octave:Braces:1:[][(){}]::2:^(top|Function)$:::2:0:
@@ -180,19 +234,17 @@
 php:String (SQL Functions):0:\\b(MATCH|AGAINST|ASCII|CHAR|SOUNDEX|MAX|MIN|MD5|LCASE|UCASE|PASSWORD|ENCRYPT|RAND|LAST_INSERT_ID|COUNT|AVG|SUM|NOW|CURDATE|CURDATE|FROM_DAYS|FROM_UNIXTIME|PERIOD_ADD|PERIOD_DIFF|TO_DAYS|UNIX_TIMESTAMP|USER|WEEKDAY|CONCAT|DATE_(FORMAT|ADD|SUB))\\b\\(:\\):1:^(PHP Block|String \\(double quoted\\)|String \\(single quoted\\))$:#999966::2:1:
 php:Operators:1:[\\+\\-\\*\\/\\.<>=`!%]+(?!\\?php)::2:^PHP Block$:#cc0000::2:1:
 php:Numbers:1:(0x)?[0-9]+(?![0-9]*[a-z\\(_\\-])::2:^PHP Block$:#330099::2:1:
-python:Keywords Definition:1:(class|def)[\\t ]+[a-zA-Z0-9_-]+(\\([^)]+\\))?[\\t ]*\:::2::#0000AA::2:0:
-python:Keywords Self:1:(self\\.?)::2::#000088::0:0:
-python:Keywords:0:\\b(and|break|continue|del|elif|else|exec|for|if|in|is|not|or|pass|print|return|while)\\b::2::#000000::2:0:
-python:Keywords Exception:0:\\b((assert|except|finally|try)[^\:]*\:|raise[^\\n]*\\n)::2::#DD0000::2:0:
-python:Comment:0:#.*?$::2::#AAAAAA::1:2:
-python:Multi-line Comment:0:''':''':1::#AAAAAA::1:2:
-python:String Double:0:":":1::#009900::0:0:
-python:String Double Escape:0:\\\\.::2:^string-double$:#009900::0:0:
-python:String Single:0:':':1::#009900::0:0:
-python:String Single Escape:0:\\\\.::2:^string-single$:#009900::0:0:
-python:Special Values:0:\\bNone\\b::2::#999900::0:0:
-python:Include:0:^(import |from ).*?$::2::#990099::0:0:
-python:Braces:0:[{()}\\[\\]]::2::#000000::2:
+python:Definition:0:(class|def)[\\t ]+[a-zA-Z0-9_-]+(\\([^)]*\\))?[\\t ]*\:::2::#0000aa::0:0:
+python:Name:1:1::3:^Definition$:#ff8800::1:0:
+python:Keywords:0:\\b(and|break|continue|del|elif|else|exec|for|from|if|import|in|is|not|or|pass|print|raise|return|while|yield|(assert|except|finally|try))\\b::2::#ff8800::0:0:
+python:Keywords Exception:0:2::3:^Keywords$:#ff8800::2:0:
+python:Keywords Self:1:self\\.?::2::#000088::0:0:
+python:Comment:1:#.*?$::2::#aaaaaa::0:0:
+python:String (double quoted):0:":":1::#009900::0:0:
+python:String (double quoted escape):0:\\\\.::2:^String \\(single quoted\\)$:#009900::0:0:
+python:String (single quoted):0:':':1::#009900::0:0:
+python:String (single quoted escape):0:\\\\.::2:^String \\(double quoted\\)$:#009900::0:0:
+python:Braces:0:[{()}\\[\\]]::2::#000000::2:0:
 R:Operators:0:[\\~\\%\\^\\&\\,\\.\\\:\\|\\+\\/\\*\\$\\!]::2::#8b2323::0:0:
 R:Assignment:1:(->>|<<-|->|<-|=|_)::2::#FF0000::2:0:
 R:Comparison:1:(<[^<-]|[^->]>|<=|>=|!=|==)::2::#8b2323::0:0:
@@ -232,7 +284,7 @@
 shell:String escape:1:\\\\.::2:^String \\([^)]+\\)$:#000099::0:0:
 shell:Variable:1:\\$[a-z][a-z0-9_]*::2:^(top|String \\([^)]+\\))$:#bb00bb::0:0:
 shell:Variable2:1:\\$(\\([^)]+\\)|{[^}]+})::2:^(top|String \\([^)]+\\))$:#bb00bb::0:0:
-shell:Variable (sys):1:\\$[\\$\\?\\*0-9]::2:^(top|String \\([^)]+\\))$:#ff0000::2:0:
+shell:Variable (sys):1:\\$[#\\$\\?\\*0-9]::2:^(top|String \\([^)]+\\))$:#ff0000::2:0:
 shell:Assignment:1:[a-z][a-z0-9_]*=::2::#424086::0:0:
 shell:Brackets:1:[\\[\\]]::2::#0000ff::2:0:
 shell:Braces:1:[{}]::2::#0000ff::2:0:
@@ -294,7 +346,7 @@
 xml:Format:1:<\\?:\\?>:1:^top$:#000000::2:1:
 xml:Tag:1:<((/)?[a-z0-9\:-]+):[^?-]?>:1::#0000ee::0:0:
 xml:Tag Name:1:1::3:^Tag$:#005676::2:0:
-xml:Attribute:1:([a-z\:\\-]*=)((?\:"[^"]*")|(?\:'[^']*'))::2:^Tag$:#3cb300::1:0:
+xml:Attribute:1:([a-z\:\\-.]*=)((?\:"[^"]*")|(?\:'[^']*'))::2:^Tag$:#3cb300::1:0:
 xml:Attribute Content:1:2::3:^Attribute$:#ff7600::1:0:
 xml:DocType:1:<!DOCTYPE [a-z0-9]+:>:1::#b35300::0:0:
 xml:Entity:1:[\\[]:[\\]]:1:^DocType$:#b35300::2:1:
